File: frontend/src/types/apiKey.ts
/**
 * API Key Management Types
 * Defines the structure for market-specific API keys used for external integrations
 */

export type ApiKeyStatus = 'active' | 'revoked';

export interface ApiKey {
  id: string;
  tenantId: string;
  marketId: string;
  name: string;           // Descriptive name for identification
  keyHash: string;        // Hashed value (never store plain text)
  lastFourChars: string;  // Last 4 characters for identification
  status: ApiKeyStatus;
  createdAt: Date;
  lastUsedAt?: Date;
  revokedAt?: Date;
  expiresAt?: Date;       // Optional expiration
  createdBy: string;      // User ID
  revokedBy?: string;     // User ID
}

export interface ApiKeyCreationResponse {
  id: string;
  key: string;            // Full key (shown only once)
  name: string;
  marketId: string;
  createdAt: Date;
}

export interface CreateApiKeyInput {
  name: string;
  expiresAt?: Date;
}

export interface ApiKeyListItem {
  id: string;
  name: string;
  lastFourChars: string;
  status: ApiKeyStatus;
  createdAt: Date;
  lastUsedAt?: Date;
  expiresAt?: Date;
}

File: api/EComm.Api/Models/ApiKey.cs
namespace EComm.Api.Models;

public class ApiKey
{
    public string Id { get; set; } = string.Empty;
    public string TenantId { get; set; } = string.Empty;
    public string MarketId { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public string KeyHash { get; set; } = string.Empty;
    public string LastFourChars { get; set; } = string.Empty;
    public string Status { get; set; } = "active"; // active, revoked
    public DateTime CreatedAt { get; set; }
    public DateTime? LastUsedAt { get; set; }
    public DateTime? ExpiresAt { get; set; }
    public string? CreatedBy { get; set; }
    public DateTime? RevokedAt { get; set; }
    public string? RevokedBy { get; set; }
}

public class ApiKeyListItem
{
    public string Id { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public string LastFourChars { get; set; } = string.Empty;
    public string Status { get; set; } = string.Empty;
    public DateTime CreatedAt { get; set; }
    public DateTime? LastUsedAt { get; set; }
    public DateTime? ExpiresAt { get; set; }
}

public class ApiKeyCreationResponse
{
    public string Id { get; set; } = string.Empty;
    public string Key { get; set; } = string.Empty; // Full key - shown only once!
    public string Name { get; set; } = string.Empty;
    public string MarketId { get; set; } = string.Empty;
    public DateTime CreatedAt { get; set; }
}

public class CreateApiKeyRequest
{
    public string Name { get; set; } = string.Empty;
    public DateTime? ExpiresAt { get; set; }
}

public class RevokeApiKeyResponse
{
    public string Message { get; set; } = string.Empty;
}
